File: Automation/001-AutomationFramework/C#/AutomationFramework/Common/Utilities/WebsiteManager.cs
using RestSharp;
using System.Net;

namespace Utilities
{
    public static class WebsiteManager
    {
        public static bool IsWebsiteAvailable(string url)
        {
            try
            {
                HttpStatusCode statusCode = new RestClient(url).ExecuteAsync(new RestRequest()).Result.StatusCode;
                return statusCode == HttpStatusCode.OK;
            }
            catch
            {
                return false;
            }
        }
    }
}

File: Automation/Woolsocks/TestAutomationFramework/src/main/java/common/utilities/WebsiteManager.java
package common.utilities;

import engines.apiTesting.RestApiManager;
import io.restassured.response.Response;
import org.apache.http.HttpStatus;
public class WebsiteManager {
    public static boolean isWebsiteAvailable(String url) {
        try {
            Response response = RestApiManager.getDataFullUrl(url);
            int statusCode = response.getStatusCode();

            return statusCode == HttpStatus.SC_OK;
        } catch (Exception exception) {
            return false;
        }
    }
}
